### ISSUE TITLE
https://community.open-emr.org/t/report-bug/9594?u=stephenwaite

### ISSUE THREAD

**Issue by stephenwaite** [openemr/openemr#1409]

**Comment by stephenwaite** [comment]
https://community.open-emr.org/t/report-bug/9594?u=stephenwaite

**Comment by sssopen** [comment]
I can't reproduce this issue. Added screenshots on the forum:
https://community.open-emr.org/t/report-bug/9594/4

**Comment by stephenwaite** [comment]
hi @sssopen , it looks like it was done with the soap note form

**Comment by sssopen** [comment]
Hi stephenwaite,
I used SOAP notes in the encounters and still can't reproduce. Added new screenshots on the forum.

**Comment by sssopen** [comment]
Thanks. I found it and was able to recreate the issue. I will look into it.

**Comment by sssopen** [comment]
Hello experts,

This seems like a design issue. Encounters are created with the user specified date so I can create an encounter with the date 04/08/18 (current date 04/04/18). However, when we add a note (ex. SOAP) to an encounter we create the note with the current timestamp (04/04/18). When we pull the encounters and notes for the report with the below SQL, the resultset is giving the data the way report is being displayed.  Do we really need to allow future dated encounters?

SELECT forms.encounter, forms.form_id, forms.form_name, 
forms.formdir, forms.date AS fdate, form_encounter.date,form_encounter.reason 
FROM forms, form_encounter 
WHERE forms.pid =1 AND form_encounter.pid = 1 AND 
form_encounter.encounter = forms.encounter AND forms.deleted=0 
ORDER BY form_encounter.encounter DESC, form_encounter.date DESC, fdate ASC

**Comment by sssopen** [comment]
Taking another look at it and the code, the data retrieved should work. Seems like there is some mess in the code.

**Comment by sssopen** [comment]
Okay. I think I found the issue. Code is expecting to have records in particular order. "New Patient Encounter" record come before the other notes in that encounter. However that not happening if you look at the SQL result:

![image](https://user-images.githubusercontent.com/5135725/38400798-f400c792-391f-11e8-93ca-a9fb33af712c.png)

Based on that resultset, the report output will be:

![image](https://user-images.githubusercontent.com/5135725/38400846-48d0b016-3920-11e8-829d-bf8f8c3a779f.png)

Either we make the  code changes to resolve this issue or we make a change in SQL to return the data the way code is expecting. So making SQL could be easier. If I change the SQL ORDER BY:

SELECT forms.encounter, forms.form_id, forms.form_name, 
forms.formdir, forms.date AS fdate, form_encounter.date,form_encounter.reason 
FROM forms, form_encounter 
WHERE forms.pid =2 AND form_encounter.pid = 2 AND 
form_encounter.encounter = forms.encounter AND forms.deleted=0 
**ORDER BY form_encounter.date DESC, form_encounter.encounter DESC**

//ORDER BY form_encounter.encounter DESC, form_encounter.date DESC, fdate ASC -- OLD

The results will be in the order the way code is expecting:

![image](https://user-images.githubusercontent.com/5135725/38400931-a38f3126-3920-11e8-836c-7c2049ca5e5c.png)

The report output will be correct, encounters will have their own notes:

![image](https://user-images.githubusercontent.com/5135725/38400950-c79df246-3920-11e8-8848-9c8ae5055ac5.png)

**Comment by stephenwaite** [comment]
really nice research and explanation @sssopen , thank you 
believe @bradymiller had suggested that we add a new feature that would enable a user to priortize the forms and how they appear in the encounter view and this could probably be extended to the report too

**Comment by bradymiller** [comment]
hi @sssopen ,

We should make sure that we don't reintroduce another bug that was fixed previously here:
https://github.com/openemr/openemr/pull/1015

-brady

**Comment by sssopen** [comment]
Hi Brady,
Thanks for letting me know about #1015. I double checked. Seems like #1015 didn't fix completely fixed the issue. If you look at the my above example closely, problem was still there that scenario. 
I tested a few different scenarios and my fix seems to be working fine.

Let me know if I should push my change.

**Comment by sssopen** [comment]
Hi Brady
Just saw your messages. Yes, I am interested working on menu customization features. Thanks for checking with me.

**Comment by bradymiller** [comment]
@sssopen , just sent you an email :)

**Comment by kacyw** [comment]
@bradymiller is this still an issue?

**Comment by bradymiller** [comment]
I don't think so, but I honestly can't figure out if it is. Per the forum thread, it appears this is a non-issue:
https://community.open-emr.org/t/same-day-multiple-encounters-patient-report-bug/9594/13

@stephenwaite , any thoughts on this?

**Comment by bradymiller** [comment]
@kacyw , forgot to mention you on above post

**Comment by stephenwaite** [comment]
guess we should try to recreate the bug to see if it addresses @vrjulas original report, @epsdky has a Fermat like comment that leaves me mystified

**Comment by kojiromike** [comment]
This was reported against OpenEMR 5.0.0 nearly 8 years ago, and maintainers were unable to confirm it was still an issue back in 2019. Closing as stale. If this is still reproducible on a current version, please reopen with updated reproduction steps.
